spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkPushedDownAggregateUtil.java

```diff
@@ -0,0 +1,373 @@
+import org.apache.iceberg.expressions.BoundAggregate;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.ExpressionVisitors;
+import org.apache.iceberg.expressions.Literal;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.spark.SparkTableUtil;
+import org.apache.iceberg.types.Conversions;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.types.Decimal;
+import scala.collection.JavaConverters;
+
+/** Helper methods for working with Spark aggregate push down. */
+public class SparkPushedDownAggregateUtil {
+
+  private SparkPushedDownAggregateUtil() {}
+
+  public static boolean metricsModeSupportsAggregatePushDown(
+      Table table, List<Expression> aggregates) {
+    MetricsConfig config = MetricsConfig.forTable(table);
+    for (Expression aggregate : aggregates) {
+      String colName = AggregateUtil.getAggregateColumnName(aggregate);
+      if (!colName.equals("*")) {
+        MetricsModes.MetricsMode mode = config.columnMode(colName);
+        if (mode.toString().equals("none")) {
+          return false;
+        } else if (mode.toString().equals("counts")) {
+          if (aggregate.op() == Expression.Operation.MAX
+              || aggregate.op() == Expression.Operation.MIN) {
+            return false;
+          }
+        } else if (mode.toString().contains("truncate")) {
+          if (AggregateUtil.getAggregateType(aggregate).typeId() == Type.TypeID.STRING) {
+            if (aggregate.op() == Expression.Operation.MAX
+                || aggregate.op() == Expression.Operation.MIN) {
+              return false;
+            }
+          }
+        }
+      }
+    }
+
+    return true;
+  }
+
+  public static InternalRow[] constructInternalRowForPushedDownAggregate(
+      SparkSession spark, Table table, List<Expression> aggregates, List<Integer> indexInTable) {
+    List<Object> valuesInSparkInternalRow = Lists.newArrayList();
+    Row[] row = SparkPushedDownAggregateUtil.getStatisticRow(spark, table);
+    for (int index = 0; index < aggregates.size(); index++) {
+      Expression aggregate = aggregates.get(index);
+      Type type = AggregateUtil.getAggregateType(aggregate);
+      valuesInSparkInternalRow.add(
+          SparkPushedDownAggregateUtil.getAggregateValue(
+              aggregate, row, indexInTable.get(index), type));
+    }
+
+    InternalRow[] rows = new InternalRow[1];
+    rows[0] = InternalRow.fromSeq(JavaConverters.asScalaBuffer(valuesInSparkInternalRow).toSeq());
+    return rows;
+  }
+
+  public static Row[] getStatisticRow(SparkSession spark, Table table) {
+    Dataset<Row> metadataRows =
+        SparkTableUtil.loadMetadataTable(spark, table, MetadataTableType.DATA_FILES);
```

Comment:
https://github.com/apache/iceberg/pull/5872#discussion_r990821627
When push down aggregates, we don't need to do a job planning because we only need a `LocalScan` on Spark driver. I guess maybe we don't need a flag to enable/disable running a parallel operation from within job planning?